homeassistant/core.py

```diff
@@ -1347,7 +1347,7 @@ def async_register(
         self,
         domain: str,
         service: str,
-        service_func: Callable,
+        service_func: Callable[[ServiceCall], Awaitable | None],
```

Comment:
```suggestion
        service_func: Callable[[ServiceCall], Awaitable[None] | None],
```

Reply:
I have adjusted it, and took the opportunity to make the same change in `async_register_admin_service`